app/queries/search_results_query.rb

```diff
@@ -7,30 +7,51 @@ class SearchResultsQuery
   CONDITION_ADWORDS_URL_CONTAINS = "array_to_string(adwords_top_urls, ', ') LIKE :word"
 
   def initialize(search_results = nil, filters = {})
-    @search_results = search_results || SearchResult.all
+    @search_results = search_results || SearchResult.order(:id).all
     @filters = filters
   end
 
   def call
     @search_results = filter_by_url(filters[:url_equals])
     @search_results = filter_adwords_url_contains(filters[:adwords_url_contains])
+    @search_results = filter_urls_contains_at_least(filters[:url_contains_at_least_one], 1)
+    @search_results = filter_urls_contains_at_least(filters[:url_contains_at_least_two], 2)
 
     @search_results
   end
 
   def filter_by_url(url)
     if url.present?
-      @search_results.where(CONDITION_URL_EQUALS, url: url)
+      @search_results.where(CONDITION_URL_EQUALS, url: url).order(:id)
     else
       @search_results
     end
   end
 
   def filter_adwords_url_contains(word)
     if word.present?
-      @search_results.where(CONDITION_ADWORDS_URL_CONTAINS, word: "%#{word}%")
+      @search_results.where(CONDITION_ADWORDS_URL_CONTAINS, word: "%#{word}%").order(:id)
     else
       @search_results
     end
   end
+
+  def filter_urls_contains_at_least(character, count = 1)
+    if character.present?
+      search_result_ids = @search_results.select do |search_result|
+        at_least_character_count?(search_result.adwords_top_urls, character, count) ||
+          at_least_character_count?(search_result.non_adwords_urls, character, count)
+      end.pluck(:id)
+
+      @search_results.where(id: search_result_ids).order(:id)
+    else
+      @search_results
+    end
+  end
+
+  private
+
+  def at_least_character_count?(urls, character, count)
+    urls.any? { |url| url.scan(character).count >= count }
```

Comment:
The test still run successfully with this update 🚀 

```suggestion
    urls.any? { |url| url.count(character) >= count }
```

Reply:
@Tuubz Can you please check this case on your side? 🤔 When using `count` instead of `scan`, it returns all the records.

![image](https://github.com/nimblehq/ic-rails-huey-toby/assets/6950766/1bdd4245-e921-481c-a064-3088733a6efd)

![image](https://github.com/nimblehq/ic-rails-huey-toby/assets/6950766/82e38845-5320-4d7c-aed2-acaf8c154d1b)

Reply:
Hmmmm, interesting...

Seems like they function differently when _more than one character_ is given 💭 

![Screen Shot 2023-07-20 at 09 17 40](https://github.com/nimblehq/ic-rails-huey-toby/assets/18277915/43267ce0-d5bb-4ae0-892d-4ae730fcf36c)

Cool, let's go with your approach 🚀